src/Gui/DlgSettingsWorkbenchesImp.cpp

```diff
@@ -0,0 +1,475 @@
+ *   This library  is distributed in the hope that it will be useful,      *
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of        *
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         *
+ *   GNU Library General Public License for more details.                  *
+ *                                                                         *
+ *   You should have received a copy of the GNU Library General Public     *
+ *   License along with this library; see the file COPYING.LIB. If not,    *
+ *   write to the Free Software Foundation, Inc., 59 Temple Place,         *
+ *   Suite 330, Boston, MA  02111-1307, USA                                *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "PreCompiled.h"
+#ifndef _PreComp_
+#include <QCheckBox>
+#include <QPushButton>
+#include <sstream>
+#endif
+
+#include "DlgSettingsWorkbenchesImp.h"
+#include "ui_DlgSettingsWorkbenches.h"
+#include "Application.h"
+#include "UserSettings.h"
+#include "Workbench.h"
+#include "WorkbenchManager.h"
+
+
+using namespace Gui::Dialog;
+
+const QString DlgSettingsWorkbenchesImp::iconLabelStr = QString::fromLatin1("iconLabel");
+const QString DlgSettingsWorkbenchesImp::nameLabelStr = QString::fromLatin1("nameLabel");
+const QString DlgSettingsWorkbenchesImp::loadLabelStr = QString::fromLatin1("loadLabel");
+const QString DlgSettingsWorkbenchesImp::loadButtonStr = QString::fromLatin1("loadButton");
+const QString DlgSettingsWorkbenchesImp::enableCheckboxStr = QString::fromLatin1("enableCheckbox");
+const QString DlgSettingsWorkbenchesImp::autoloadCheckboxStr = QString::fromLatin1("autoloadCheckbox");
+
+/* TRANSLATOR Gui::Dialog::DlgSettingsWorkbenchesImp */
+
+/**
+ *  Constructs a DlgSettingsWorkbenchesImp
+ */
+DlgSettingsWorkbenchesImp::DlgSettingsWorkbenchesImp( QWidget* parent )
+    : PreferencePage( parent )
+    , ui(new Ui_DlgSettingsWorkbenches)
+{
+    ui->setupUi(this);
+    connect(ui->AutoloadModuleCombo, QOverload<int>::of(&QComboBox::activated), this, [this](int index) { onStartWbChangedClicked(index); });
+}
+
+/**
+ *  Destroys the object and frees any allocated resources
+ */
+DlgSettingsWorkbenchesImp::~DlgSettingsWorkbenchesImp()
+{
+}
+
+
+void DlgSettingsWorkbenchesImp::saveSettings()
+{
+    std::ostringstream enabledStr, disabledStr, autoloadStr;
+
+    for (int i = 0; i < ui->wbList->count(); i++) {
+        QWidget* widget = ui->wbList->itemWidget(ui->wbList->item(i));
+        if (!widget)
+            continue;
+        QCheckBox* enableBox = widget->findChild<QCheckBox*>(enableCheckboxStr);
+        if (!enableBox)
+            continue;
+
+        if (enableBox->isChecked()) {
+            if (!enabledStr.str().empty())
+                enabledStr << ",";
+            enabledStr << widget->objectName().toStdString();
```

Comment:
You could advantageously use a lambda for these cases.

Reply:
Done